lib/rules/selector-max-attribute/index.mjs

```diff
@@ -92,8 +90,21 @@ const rule = (primary, secondaryOptions) => {
 			if (!isStandardSyntaxRule(ruleNode)) return;
 
 			flattenNestedSelectorsForRule(ruleNode, result).forEach(({ selector, resolvedSelectors }) => {
-				resolvedSelectors.forEach((resolvedSelector) => {
-					checkSelector(resolvedSelector, selector, ruleNode);
+				resolvedSelectors.walk((childSelector) => {
+					if (childSelector.type !== 'selector') return;
+
+					if (isRoot(childSelector.parent)) {
+						checkSelector(childSelector, selector, ruleNode);
+
+						return;
+					}
+
+					if (
+						isPseudo(childSelector.parent) &&
+						isContextFunctionalPseudoClass(childSelector.parent)
+					) {
```

Comment:
[suggest] How about simplifying callers of `isContextFunctionalPseudoClass`?

```suggestion
					if (isContextFunctionalPseudoClass(childSelector.parent)) {
```

Perhaps we could achieve this by changing the parameter type of `isContextFunctionalPseudoClass` like this:

```diff
- * @param {import('postcss-selector-parser').Node} node - postcss-selector-parser node (of type pseudo)
+ * @param {import('postcss-selector-parser').Node | import('postcss-selector-parser').Container | undefined} node - postcss-selector-parser node (of type pseudo)
```

Reply:
This also made it possible to combine the two conditions (`isRoot` and `isContextFunctionalPseudoClass`) in a single `if` while keep it readable.

Let me know what you think 🙇

Reply:
Great! 👏🏼